File: moments-pipeline-sim/apps/api-ts/src/types/moment.ts
export type MomentType =
  | "goal"
  | "lead_change"
  | "timeout"
  | "clutch_play"
  | "milestone";

export interface MomentEvent {
  event_id: string;
  event_time: string;
  ingest_time?: string;
  league: string;
  game_id: string;
  team_id?: string;
  player_id?: string;
  moment_type: MomentType;
  importance_score: number;
  metadata: Record<string, unknown>;
}

File: moments-pipeline-sim/apps/consumer-rust/src/moment.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct MomentEvent {
    pub event_id: String,
    pub event_time: String,
    pub ingest_time: Option<String>,
    pub league: String,
    pub game_id: String,
    pub team_id: Option<String>,
    pub player_id: Option<String>,
    pub moment_type: String,
    pub importance_score: f64,
    pub metadata: Value,
}
